docs/jwst/outlier_detection/outlier_detection.rst

```diff
@@ -50,6 +50,55 @@ Specifically, this routine performs the following operations:
 * Perform statistical comparison between blotted image and original image to identify outliers.
 * Update input data model DQ arrays with mask of detected outliers.
 
+Memory Model for Outlier Detection Algorithm
+---------------------------------------------
+The outlier detection algorithm can end up using massive amounts of memory
+depending on the number of inputs, the size of each input, and the size of the
+final output product.  Specifically,
+
+    * The input :py:class:`~jwst.datamodels.ModelContainer` or
+      :py:class:`~jwst.datamodels.CubeModel`
+      for IFU data, by default, all input exposures would have been kept open in memory to make
+      processing more efficient.
+
+    * The initial resample step creates an output product for EACH input that is the
+      same size as the final
+      output product, which for imaging modes can span all chips in the detector while
+      also accounting for all dithers.  For some Level 3 products, each resampled image can
+      be on the order of 2Gb or more.
+
+    * The median combination step then needs to have all pixels at the same position on
+      the sky in memory in order to perform the median computation.  The simplest implmentation
+      for this step would require keeping all resampled outputs in memory at the same time.
```

Comment:
I'm not sure I'd say "would require". It *does* require keeping all outputs in memory and that's the way the initial implementation has already been done, which is still available to use (it is still available, right?). So I would say "... for this step requires keeping all ..."

Reply:
Revised under 8280daf.